include/audio/aseq.h

```diff
@@ -0,0 +1,1979 @@
+    .elseif ASEQ_MODE == ASEQ_MODE_TABLE && \tbl_id != -1
+        _check_arg_bitwidth_u \tbl_id, 8
+        .byte (\tbl_id & 0xFF) | (\lo_bits)
+    .elseif ASEQ_MODE == ASEQ_MODE_ARRAY && \arr_id != -1
+        _check_arg_bitwidth_u \arr_id, 8
+        .byte (\arr_id & 0xFF) | (\lo_bits)
+    .elseif ASEQ_MODE == ASEQ_MODE_FILTER && \filt_id != -1
+        _check_arg_bitwidth_u \filt_id, 8
+        .byte (\filt_id & 0xFF) | (\lo_bits)
+    .elseif ASEQ_MODE == ASEQ_MODE_ENVELOPE && \env_id != -1
+        _check_arg_bitwidth_u \env_id, 8
+        .byte (\env_id & 0xFF) | (\lo_bits)
+    .elseif ASEQ_MODE == ASEQ_MODE_BUFFER && \buf_id != -1
+        _check_arg_bitwidth_u \buf_id, 8
+        .byte (\buf_id & 0xFF) | (\lo_bits)
+    .else
+        _section_invalid \cmd_name
+    .endif
+.endm
+
+.macro .ptr value
+    _wr_lbl \value
+.endm
+
+.macro .ptr_raw value
+    _wr16 \value
+.endm
+
+
+
+/**
+ *  Instructions
+ */
+
+/**
+ * end
+ *
+ *  Marks the end of an executable section.
+ */
+.macro end
+    _wr_cmd_id  end, 0xFF,0xFF,0xFF,,,,,, 0, 0
+.endm
+
+/**
+ * delay1
+ *
+ *  Delays for one tick.
+ */
+.macro delay1
+    _wr_cmd_id  delay1, 0xFE,0xFE,,,,,,, 0, 0
+.endm
+
+/**
+ * delay <delay:var>
+ *
+ *  Delays for `delay` ticks.
+ */
+.macro delay delay
+    _wr_cmd_id  delay, 0xFD,0xFD,,,,,,, 0, 0
+    _var        \delay
+.endm
+
+/**
+ * call <label:lbl>
+ *
+ *  Unconditionally enters a subroutine at `label`. Execution will resume following this instruction once the
+ *  subroutine encounters an `end` instruction.
+ */
+.macro call label
+    _wr_cmd_id  call, 0xFC,0xFC,0xFC,,,,,, 0, 0
+    _wr_lbl     \label
+.endm
+
+/**
+ * jump <label:lbl>
+ *
+ *  Branches to `label` unconditionally.
+ */
+.macro jump label
+    _wr_cmd_id  jump, 0xFB,0xFB,0xFB,,,,,, 0, 0
+    _wr_lbl     \label
+.endm
+
+/**
+ * beqz <label:lbl>
+ *
+ *  Branches to `label` if TR == 0.
+ */
+.macro beqz label
+    _wr_cmd_id  beqz, 0xFA,0xFA,0xFA,,,,,, 0, 0
+    _wr_lbl     \label
+.endm
+
+/**
+ * bltz <label:lbl>
+ *
+ *  Branches to `label` if TR < 0.
+ */
+.macro bltz label
+    _wr_cmd_id  beqz, 0xF9,0xF9,0xF9,,,,,, 0, 0
+    _wr_lbl     \label
+.endm
+
+/**
+ * loop <num:u8>
+ *
+ *  Begin a loop for `num` iterations.
+ *
+ *  Maximum loop nesting level is 4 - callDepth, where callDepth is the
+ *  current subroutine nesting level.
+ */
+.macro loop num
+    _wr_cmd_id  loop, 0xF8,0xF8,0xF8,,,,,, 0, 0
+    _wr_u8      \num
+.endm
+
+/**
+ * loopend
+ *
+ *  Indicates the end of a loop body. Whenever the instruction is executed
+ *  the loop counter decrements and either branches back to the loop start
+ *  if there are remaining iterations or exits the loop if all iterations
+ *  have been completed.
+ */
+.macro loopend
+    _wr_cmd_id  loopend, 0xF7,0xF7,0xF7,,,,,, 0, 0
+.endm
+
+/**
+ * break
+ *
+ *  Immediately ends the current loop, but does not modify PC.
+ *  Do not allow execution to later flow into loopend.
```

Comment:
I can't tell if this is a description of what this command does, or a warning to the user of what not to do